### PR TITLE
Unified Storage: Adds sql metrics

### PR DESCRIPTION
**Pull request by owensmallwood**
Since we use a separate db engine for US, we weren't getting any db metrics.

### CODE REVIEW

File: pkg/services/store/entity/db/dbimpl/dbimpl.go [hunk 1]
```diff
@@ -69,7 +71,7 @@ func (db *EntityDB) GetEngine() (*xorm.Engine, error) {
 			}
 
 			connectionString := fmt.Sprintf(
-				"user='%s' password='%s' host='%s' port='%s' dbname='%s' sslmode='%s'", // sslcert='%s' sslkey='%s' sslrootcert='%s'",
+				"user=%s password=%s host=%s port=%s dbname=%s sslmode=%s", // sslcert='%s' sslkey='%s' sslrootcert='%s'",
```
Comment (owensmallwood) [inline pkg/services/store/entity/db/dbimpl/dbimpl.go:74]: Had to change this so things can run locally. This issue is fixed [in this PR](https://github.com/grafana/grafana/pull/87656)

File: pkg/services/store/entity/db/dbimpl/dbimpl.go [hunk 2]
```diff
@@ -138,6 +140,11 @@ func (db *EntityDB) GetEngine() (*xorm.Engine, error) {
 		return nil, err
 	}
 
+	// register the go_sql_stats_connections_* metrics
+	if err := prometheus.Register(sqlstats.NewStatsCollector("unified_storage", db.engine.DB().DB)); err != nil {
+		db.log.Warn("Failed to register unified storage sql stats collector", "error", err)
+	}
+
```
Comment (DanCech) [inline pkg/services/store/entity/db/dbimpl/dbimpl.go:147]: Should this be up in the block where we create the new connection? Or do we need it even when we are piggybacking on the main grafana connection?

Comment (owensmallwood) [reply pkg/services/store/entity/db/dbimpl/dbimpl.go:147]: If we're piggybacking on the grafana connection, then this function should return right away bc of this
```go
if db.engine != nil {
    return db.engine, nil
}
```
at the beginning.

Comment (DanCech) [reply pkg/services/store/entity/db/dbimpl/dbimpl.go:147]: nope, check out lines 128-134

Comment (DanCech) [reply pkg/services/store/entity/db/dbimpl/dbimpl.go:147]: the check for whether we've already set up db.engine is just a shortcut so you can call GetEngine again and it'll return the engine we configured the first time.

Comment (owensmallwood) [reply pkg/services/store/entity/db/dbimpl/dbimpl.go]: ah! gotcha. Fixed it 👍

**Review by DanCech — APPROVED** [review]

**Review by inf0rmer — APPROVED** [review]

**Pull request merged**